File: modules/swagger-jersey2-jaxrs/src/main/scala/com/wordnik/swagger/jersey/config/JerseyJaxrsConfig.scala
package com.wordnik.swagger.jersey.config

import com.wordnik.swagger.jaxrs.config._

import com.wordnik.swagger.jersey.JerseyApiReader
import com.wordnik.swagger.config.{ ConfigFactory, ScannerFactory }
import com.wordnik.swagger.reader._

import javax.servlet.ServletConfig
import javax.servlet.http.HttpServlet

class JerseyJaxrsConfig extends HttpServlet {
  override def init(servletConfig: ServletConfig) = {
    super.init(servletConfig)

    implicit val config = servletConfig
    ConfigFactory.config = new WebXMLReader()
    ScannerFactory.scanner = Some(new DefaultJaxrsScanner())
    ClassReaders.reader = Some(new JerseyApiReader)
  }
}

File: modules/swagger-jersey2-jaxrs/src/main/java/com/wordnik/swagger/jersey/config/JerseyJaxrsConfig.java
package com.wordnik.swagger.jersey.config;

import com.wordnik.swagger.jaxrs.config.*;

import javax.ws.rs.core.*;

import javax.servlet.ServletConfig;
import javax.servlet.http.HttpServlet;

public class JerseyJaxrsConfig extends HttpServlet {
  @Override
  public void init(ServletConfig servletConfig) throws javax.servlet.ServletException {
    super.init(servletConfig);

    servletConfig.getServletContext().setAttribute("reader", new WebXMLReader(servletConfig));
    servletConfig.getServletContext().setAttribute("scanner", new DefaultJaxrsScanner());
    ReaderConfigUtils.initReaderConfig(servletConfig);
  }
}
